src/libraries/System.IO.Pipes/src/System/IO/Pipes/PipeStream.ValueTaskSource.cs

```diff
@@ -75,11 +75,11 @@ internal void RegisterForCancellation(CancellationToken cancellationToken)
                         _cancellationRegistration = cancellationToken.UnsafeRegister(static (s, token) =>
                         {
                             PipeValueTaskSource vts = (PipeValueTaskSource)s!;
-                            if (!vts._pipeStream.SafePipeHandle.IsInvalid)
+                            if (vts._pipeStream.InternalHandle is { IsInvalid: false } handle)
```

Comment:
A closed `SafeFileHandle` returns `false` for `IsInvalid`. I know that we have a `catch (ObjectDisposedException)` block here, but I think that it would be better to check for `IsClosed` instead.

```suggestion
                            if (vts._pipeStream.InternalHandle is { IsClosed: false } handle)
```

<details>

```cs
FileStream fs = File.Create("test.txt");
Microsoft.Win32.SafeHandles.SafeFileHandle safeFileHandle = fs.SafeFileHandle;
fs.Dispose();
Console.WriteLine(safeFileHandle.IsClosed);
Console.WriteLine(safeFileHandle.IsInvalid);
```

```log
dotnet run -c Release
True
False
```

</details>